src/release-specification.ts

```diff
@@ -301,6 +313,55 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release new versions of dependencies alongside their dependents
+      if (
+        pkg &&
+        versionSpecifierOrDirective &&
+        (hasProperty(IncrementableVersionParts, versionSpecifierOrDirective) ||
+          isValidSemver(versionSpecifierOrDirective))
+      ) {
+        const missingDependencies = Object.keys({
+          ...(pkg.unvalidatedManifest.dependencies || {}),
+          ...(pkg.unvalidatedManifest.peerDependencies || {}),
```

Comment:
Note that if you incorporate `dependencies` and `peerDependencies` into the `validatedManifest` then you can default these properties to an empty object, so then you could simplify this to:

``` suggestion
          ...pkg.validatedManifest.dependencies,
          ...pkg.validatedManifest.peerDependencies,